airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/PersistStateActivityImpl.java

```diff
@@ -34,15 +41,23 @@ public boolean persist(final UUID connectionId, final StandardSyncOutput syncOut
       try {
         final Optional<StateWrapper> maybeStateWrapper = StateMessageHelper.getTypedState(state.getState(), featureFlags.useStreamCapableState());
         if (maybeStateWrapper.isPresent()) {
-          final Optional<StateWrapper> previousState = statePersistence.getCurrentState(connectionId);
-          final StateType newStateType = maybeStateWrapper.get().getStateType();
-          if (statePersistence.isMigration(connectionId, newStateType, previousState) && newStateType == StateType.STREAM) {
-            validateStreamStates(maybeStateWrapper.get(), configuredCatalog);
+          final ConnectionState previousState = airbyteApiClient.getConnectionApi()
+              .getState(new ConnectionIdRequestBody().connectionId(connectionId));
+          if (previousState != null) {
```

Comment:
I suspect this should be part of the API (as in StateHandler).
From the definition, we are exposing an API to update states, I think those check should happen regardless of the path we take since they are here to ensure or states remain consistent.
I would expect PersistState to only format the query to call the API.

Reply:
Continued in child PR here: https://github.com/airbytehq/airbyte/pull/15902/files#r953198775